### PR TITLE
WIP: add symbol versioning

### PR DESCRIPTION
**Pull request by p-steuer**
Work In Progress: add symbol versioning

Signed-off-by: Patrick Steuer <patrick.steuer@de.ibm.com>

### CODE REVIEW

File: libica.map [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+	ica_sha512;
+	ICA_DRBG_SHA512;
+	s390_get_functionlist;
+	s390_initialize_functionlist;
+	_fini;
+	_init;
+    local: *;
+};
+
+LIBICA_3.0.0_FIPS {
```
Comment (sharkcz) [inline libica.map]: nice, now even FIPS and non-FIPS variants can be detected

File: libica.map [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+	ica_sha3_224;
+	ica_sha3_256;
+	ica_sha3_384;
+	ica_sha3_512;
+	ica_shake_128;
+	ica_shake_256;
+    local: *;
+} LIBICA_3.0.0;
+
+LIBICA_3.1.0_FIPS {
```
Comment (sharkcz) [inline libica.map]: isn't this redundant (covered already by LIBICA_3.0.0_FIPS)? or are the "fips" functions with incompatible ABI in 3.1?

Comment (p-steuer) [reply libica.map]: youre right, this is redundant.

**Comment by hfreude** [comment]
Nice work, approved.

**Pull request merged**